src/ui/preview/preview.js

```diff
@@ -463,11 +473,81 @@ cwc.ui.Preview.prototype.focus = function() {
 /**
  * Injects and executes the passed code in the preview content, if supported.
  * @param {!(string|Function)} code
+ * @param {Number} timeout
+ * @return {!Promise}
+ * @export
  */
-cwc.ui.Preview.prototype.executeScript = function(code) {
+cwc.ui.Preview.prototype.executeScript = function(code, timeout = 500) {
```

Comment:
Did you try if an value with 200 worked as well ?
During my test in most cases the preview was loaded in 0.02 - 0.2 secs.

Reply:
I didn't. 500 is just a mildly educated guess. I'm not sure how to test for the correct value, since the load time on my beefly workstations may differ from the underpowered chromebooks of the typical users. Half a second seems like enough headroom for just about any system and it's better to have a timeout that's too long than too short. The promise will return whenever the executed code completes. The timeout simply rejects the promise if takes longer to complete than the timeout value.

Also, in my testing and reading, it seems that iframes don't execute in a separate thread like webviews do. This renders the timeout ineffective for iframes -- it will still reject if the code took longer than the timeout, but it won't do so until the code completes. While I don't have a solution for that problem (which I think could potentially result in executed code hanging all of CwC when iframes are in use), I do have a WIP that will use Worker instead of Webview to test the multi-threaded case in tests for cwc.Messenger.